### PR TITLE
feat: update to Nx11

### PR DESCRIPTION
**Pull request by nartc**
This PR proposes the following changes:
- updates the repo to Nx 11, AngularCLI 11, and Angular Schematics v11
- adds `checkCliVersion` to check for AngularCLI 11.2
- skips unit tests as there are bugs in the Nrwl schematics itself.
- skips e2e tests as the `nrwl/angular` schematics installs Angular v11.0.7 and will not pass Angular CLI version check

The tests are skipped but the schematics have been verified locally.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npx lint-staged
```
Comment (vltansky) [inline .husky/pre-commit:4]: Would we maybe want to use [simple-git-hooks](https://github.com/toplenboren/simple-git-hooks) instead of husky?

**Review by vltansky — APPROVED** [review]

**Review by vltansky — CHANGES_REQUESTED** [review]
```
$ npm run build

> @ngneat/tailwind@6.0.3 build D:\projects\tailwind
> ng run tailwind:build-copy


>  NX   ERROR  Cannot find target 'build-copy' for project 'tailwind'

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ngneat/tailwind@6.0.3 build: `ng run tailwind:build-copy`
npm ERR! Exit status 1
```

changing to `nx run tailwind:build-copy` fixes it for me

**Comment by vltansky** [comment]
```
$ npm run test

> tailwind-test@0.0.0 test D:\projects\tailwind-test
> npx ng generate ./../tailwind/dist/libs/tailwind:ng-add

? Would you like to enable dark mode? none
? What @tailwindcss plugins do you want to enable?

    Detected AngularCLI version is 11.1.4 which does not support TailwindCSS natively.
    Please run "ng add @ngneat/tailwind@6" for Custom Webpack support.

Nothing to be done.
```

Version check works :)

File: .husky/pre-commit (hunk 1 above)
Comment (nartc) [reply .husky/pre-commit:4]: Will do!

**Pull request merged**